instrumentation/apache-dubbo-2.7/library-autoconfigure/src/main/java/io/opentelemetry/instrumentation/apachedubbo/v2_7/TracingFilter.java

```diff
@@ -18,14 +18,52 @@
 
 final class TracingFilter implements Filter {
 
-  private final Instrumenter<DubboRequest, Result> clientInstrumenter;
-  private final Instrumenter<DubboRequest, Result> serverInstrumenter;
-
-  TracingFilter(
-      Instrumenter<DubboRequest, Result> clientInstrumenter,
-      Instrumenter<DubboRequest, Result> serverInstrumenter) {
-    this.clientInstrumenter = clientInstrumenter;
-    this.serverInstrumenter = serverInstrumenter;
+  private final InstrumenterSupplier instrumenterSupplier;
+
+  private TracingFilter(InstrumenterSupplier instrumenterSupplier) {
+    this.instrumenterSupplier = instrumenterSupplier;
+  }
+
+  static TracingFilter newClientFilter(Instrumenter<DubboRequest, Result> clientInstrumenter) {
+    return newFilter(clientInstrumenter, true);
+  }
+
+  static TracingFilter newServerFilter(Instrumenter<DubboRequest, Result> serverInstrumenter) {
+    return newFilter(serverInstrumenter, false);
+  }
+
+  private static TracingFilter newFilter(
+      Instrumenter<DubboRequest, Result> instrumenter, boolean isClientSide) {
+    return new TracingFilter(
+        new InstrumenterSupplier() {
+
+          @Override
+          public Instrumenter<DubboRequest, Result> get(RpcContext rpcContext) {
+            return instrumenter;
+          }
+
+          @Override
+          public boolean isClientSide(RpcContext rpcContext) {
+            return isClientSide;
+          }
+        });
+  }
+
+  static TracingFilter newFilter(
+      Instrumenter<DubboRequest, Result> serverInstrumenter,
+      Instrumenter<DubboRequest, Result> clientInstrumenter) {
+    return new TracingFilter(
+        new InstrumenterSupplier() {
+          @Override
+          public Instrumenter<DubboRequest, Result> get(RpcContext rpcContext) {
+            return rpcContext.isConsumerSide() ? clientInstrumenter : serverInstrumenter;
```

Comment:
Do you think we should keep as it is? Even though it might be wrong in some scenarios?